flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/flink/FlinkDistribution.java

```diff
@@ -301,7 +309,105 @@ public void submitSQLJob(SQLJobSubmission job, Duration timeout) throws Exceptio
                             });
             new Thread(future).start();
             future.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+        } else if (job.getClientMode() == SQLJobSubmission.ClientMode.REST) {
+            FutureTaskWithException<Void> future =
+                    new FutureTaskWithException<>(
+                            () -> {
+                                Map<String, String> configMap =
+                                        GlobalConfiguration.loadConfiguration(
+                                                        conf.toAbsolutePath().toString())
+                                                .toMap();
+                                String host =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.address",
+                                                InetAddress.getByName("localhost")
+                                                        .getHostAddress());
+                                String port =
+                                        configMap.getOrDefault(
+                                                "sql-gateway.endpoint.rest.port", "8083");
+                                // Open a session
+                                String sessionHandle = openSession(host, port);
```

Comment:
Here missing the logic about the ADD JAR. Please cc the logic in the Hive_JDBC

Reply:
Done.